src/RelSeries.jl

```diff
@@ -1043,32 +1043,72 @@ end
 #
 ###############################################################################
 
-@doc Markdown.doc"""
-    sqrt(a::RelSeriesElem)
-
-Return the square root of the power series $a$. By default the function raises
-an exception if the input is not a square. If `check=false` this check is
-omitted.
-"""
-function Base.sqrt(a::RelSeriesElem; check::Bool=true)
+function sqrt_classical_char2(a::RelSeriesElem; check::Bool=true)
+   S = parent(a)
+   R = base_ring(a)
+   prec = div(precision(a) + 1, 2)
+   if iszero(a)
+      asqrt = parent(a)()
+      asqrt = set_precision!(asqrt, prec)
+      asqrt = set_valuation!(asqrt, prec)
+      return true, asqrt
+   end
    aval = valuation(a)
-   !iseven(aval) && error("Not a square in sqrt")
+   if check && !iseven(aval)
+      return false, S()
+   end
+   aval2 = div(aval, 2)
+   asqrt = parent(a)()
+   fit!(asqrt, prec)
+   asqrt = set_precision!(asqrt, prec)
+   asqrt = set_valuation!(asqrt, aval2)
+   if check
+      for i = 1:2:precision(a) - aval - 1 # series must have even exponents
+         if !iszero(polcoeff(a, i))
+            return false, S()
+         end
+      end
+   end
+   for i = 0:prec - aval2 - 1
+      c = polcoeff(a, 2*i)
+      if check && !issquare(c)
+         return false, S()
+      end
+      asqrt = setcoeff!(asqrt, i, sqrt(c; check=false))
+   end
+   asqrt = set_length!(asqrt, normalise(asqrt, prec))
+   return true, asqrt
+end
+
+function sqrt_classical(a::RelSeriesElem; check::Bool=true)
    R = base_ring(a)
+   S = parent(a)
+   aval = valuation(a)
+   if check && !iseven(aval)
+      return false, S()
+   end
    !isdomain_type(elem_type(R)) && error("Sqrt not implemented over non-integral domains")
+   if characteristic(R) == 2
+      return sqrt_classical_char2(a, check=check)
+   end
    aval2 = div(aval, 2)
    prec = precision(a) - aval
    if prec == 0
       asqrt = parent(a)()
       asqrt = set_precision!(asqrt, aval2)
       asqrt = set_valuation!(asqrt, aval2)
-      return asqrt
+      return true, asqrt
```

Comment:
There are too many untested corner cases like this. Add some test cases to make sure that these bits of code are actually doing the correct thing. Code coverage actually decreases with the current state of this PR.